ipq806x: Increase kernel partition size to 4 MB for EA8500/EA7500v1

hnyman opened the pull request:
Increase the kernel size from 3 MB to 4 MB for EA8500 and EA7500v1 to enable building them in master.
* modify the common .dtsi
* modify the kernel size in the image recipes
* Define compat-version 2.0 to force factory image usage for sysupgrade. Add explanation message.

Reference to forum discussion at 
https://forum.openwrt.org/t/increasing-ea8500-ea7500-kernel-size-to-enable-builds/121039

cc @chunkeey

pepe2k commented:
Out of curiosity, why not extend it lets say twice, to 6 MB? Wouldn't that survive longer and more kernel updates/size increases?
Is there a limit of data size U-Boot reads from kernel partition?

Cheers,
Piotr

hnyman commented:
I didn't research further than the 4 MB that has been earlier set as the new enlarged kernel size for many other ipq806x devices. (like R7800 etc.)

hnyman commented:
Successful testing reports at 
https://forum.openwrt.org/t/increasing-ea8500-ea7500-kernel-size-to-enable-builds/121039/23

Cc @chunkeey

mans0n commented:
> I didn't research further than the 4 MB that has been earlier set as the new enlarged kernel size for many other ipq806x devices. (like R7800 etc.)

Could you point out a recent kernel size increase? The most recent one I found is c3af761e4740 in 2018, which is too old for a reference IMO.

hnyman commented:
> Could you point out a recent kernel size increase? 

I referenced them in the forum thread I linked. R7800 & related was the first major one, and then recently mvebu wrt1900acv1 and wrt32 with
https://github.com/openwrt/openwrt/commit/10415d5e7016b69dc71c5f1b03e8e17b586f8edd

hnyman commented on target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8064-eax500.dtsi:
```diff
@@ -133,12 +133,12 @@
+				reg = <0x0f80000 0x2800000>;  /* 4 MB, spill to rootfs */
 			};
 
 			partition@1280000 {
```
you are right, I missed the section label change.
Will fix it today.

hnyman replied:
@mans0n 
I fixed  both section labels

mans0n commented:
Thanks. dc50694bd1a8 was for 4.14 and mvebu is a different target...
I don't object the 4MB change but I agree that 6MB would be more future-proof as @pepe2k said.

hnyman committed:
ipq806x: Increase kernel size to 4 MB for EA8500/EA7500v1

Increase the kernel size from 3 MB to 4 MB for EA8500 and
EA7500v1.
* modify the common .dtsi
* modify the kernel size in the image recipes

Define compat-version 2.0 to force factory image usage for sysupgrade.
Add explanation message.

Signed-off-by: Hannu Nyman <hannu.nyman@iki.fi>

hnyman committed:
Revert "ipq806x: disable Linksys EA7500v1 and EA8500"

This reverts commit 5696244ae242b7965b6754a59aa104016840db52.

As the kernel size for EA8500 and EA7500v1 has been increased
to 4 MB, the builds can be enabled again.

Signed-off-by: Hannu Nyman <hannu.nyman@iki.fi>

chunkeey commented:
so `bootipq` can be suspect at times. I discovered this with a ipq40xx device ( EZVIZ CS-W3-WD1200G EUP) and it had a real 4MiB limit and a new bootcmd was needed 45eb57f12f3a128a1. 

Unfortunately Linksys didn't include the uboot sources in their GPL drops. So, we can only guess..

As for the EA7500-v1 sysupgrade I downloaded from the dropbox: It has a 3053870 Byte (= 2982 KiB) kernel.
Can you please enable CONFIG_ALL_KMODS? This would hopefully push the kernel beyond the 3072KiB and would confirm
that just enlarging the kernel partition is enough and the limit isn't at 3072K.

chunkeey commented:
A +1 MiB kernel partition size increase can be misleading (as in: it looks smaller than it is).
The devices use zImage. This means that the kernel is xz compressed with a decompressor as a loader.

Todays OpenWrts' IPQ806x R7500 kernel sits at 3200KiB and "unpacked" it's 11.6MiB (~ 3.7 compression ratio.)
If this 3.7 compression ratio holds "true" for the future. A 4096 KiB zImage could be as big as 15 MiB uncompressed.
A 5120 KiB zImage would be around 18 MiB and 6144 KiB would be in the 22 MiB neighborhood.
